commit 64df73c2ea0e02f7794ba2f08540a74a9bb198db
Author: Siddhesh Poyarekar <siddhesh@redhat.com>
Date:   Sat Jun 28 06:05:24 2014 +0530

    Fix Wundef warning for MEMCPY_OK_FOR_FWD_MEMMOVE
    
    Define MEMCPY_OK_FOR_FWD_MEMMOVE in memcopy.h and let arch-specific
    implementations of that file override the value if necessary.  This
    override is only useful for tile and moving this macro to memcopy.h
    allows us to remove the tile-specific memmove.c.

Conflicts:
	sysdeps/tile/tilegx/memmove.c

Due to copyright header change; file removed manually.

diff --git a/sysdeps/generic/memcopy.h b/sysdeps/generic/memcopy.h
index 631214b85ab81d57..08892a4ea33f1ca7 100644
--- a/sysdeps/generic/memcopy.h
+++ b/sysdeps/generic/memcopy.h
@@ -148,4 +148,6 @@ extern void _wordcopy_bwd_dest_aligned (long int, long int, size_t) __THROW;
 /* Threshold value for when to enter the unrolled loops.  */
 #define	OP_T_THRES	16
 
+#define MEMCPY_OK_FOR_FWD_MEMMOVE 0
+
 #endif /* memcopy.h */
diff --git a/sysdeps/tile/memcopy.h b/sysdeps/tile/memcopy.h
index 2bc3fce6864315b8..2876baa0e9f9ab31 100644
--- a/sysdeps/tile/memcopy.h
+++ b/sysdeps/tile/memcopy.h
@@ -19,6 +19,12 @@
 #include <sysdeps/generic/memcopy.h>
 #include <bits/wordsize.h>
 
+/* The tilegx implementation of memcpy is safe to use for memmove.  */
+#ifdef __tilegx__
+# undef MEMCPY_OK_FOR_FWD_MEMMOVE
+# define MEMCPY_OK_FOR_FWD_MEMMOVE 1
+#endif
+
 /* Support more efficient copying on tilegx32, which supports
    long long as a native 64-bit type.  */
 #if defined (__tilegx__) && __WORDSIZE == 32
diff --git a/sysdeps/tile/tilegx/memmove.c b/sysdeps/tile/tilegx/memmove.c
deleted file mode 100644
index 0cb0a81e5fd53639..0000000000000000
--- a/sysdeps/tile/tilegx/memmove.c
+++ /dev/null
@@ -1,22 +0,0 @@
-/* Copy memory to memory until the specified number of bytes
-   has been copied.  Overlap is handled correctly.
-   Copyright (C) 2012 Free Software Foundation, Inc.
-   This file is part of the GNU C Library.
-
-   The GNU C Library is free software; you can redistribute it and/or
-   modify it under the terms of the GNU Lesser General Public
-   License as published by the Free Software Foundation; either
-   version 2.1 of the License, or (at your option) any later version.
-
-   The GNU C Library is distributed in the hope that it will be useful,
-   but WITHOUT ANY WARRANTY; without even the implied warranty of
-   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
-   Lesser General Public License for more details.
-
-   You should have received a copy of the GNU Lesser General Public
-   License along with the GNU C Library; if not, see
-   <http://www.gnu.org/licenses/>.  */
-
-/* The tilegx implementation of memcpy is safe to use for memmove.  */
-#define MEMCPY_OK_FOR_FWD_MEMMOVE 1
-#include <string/memmove.c>
